test/test_broker/iaas.AzureClient.spec.js

```diff
@@ -42,6 +42,182 @@ describe('iaas', function () {
+          zones: [zone],
+          location: loc,
+          tags: {
+            name: 'value',
+            createdBy: 'service-fabrik'
+          },
+          sku: {
+            name: 'Premium_LRS'
+          },
+          properties: {
+            creationData: {
+              createOption: 'Copy',
+              sourceUri: 'testId'
+            }
+          }
+        }, null, 'diskfailed');
+        return client.createDiskFromSnapshot(snapshotName, zone, {
+            sku: sku,
+            tags: {
+              name: 'value'
+            }
+          })
+          .catch((err) => {
+            expect(err.message).to.equal('diskfailed');
+            mocks.verify();
+          });
+      });
+
+      it('should create disk from snapshot successfully', function () {
+        mocks.azureClient.auth(2);
+        mocks.azureClient.getSnapshot(`/subscriptions/${settings.subscription_id}/resourceGroups/${settings.resource_group}/providers/Microsoft.Compute/snapshots/${snapshotName}?api-version=2017-03-30`, {
+          status: 200,
+          body: {
+            id: 'testId',
+            location: 'westeurope'
+          },
+          headers: {
+            'x-ms-request-id': '774c96e7-0001-0006-7e01-67617f000000',
+            'x-ms-version': '2016-05-31',
+            date: new Date().toISOString()
+          }
+        });
+
+        mocks.azureClient.createDisk(`/subscriptions/${settings.subscription_id}/resourceGroups/${settings.resource_group}/providers/Microsoft.Compute/disks/${diskName}?api-version=2017-03-30`, {
+          zones: [zone],
+          location: loc,
+          tags: {
+            name: 'value',
+            createdBy: 'service-fabrik'
+          },
+          sku: {
+            name: 'Premium_LRS'
+          },
+          properties: {
+            creationData: {
+              createOption: 'Copy',
+              sourceUri: 'testId'
+            }
+          }
+        }, {
+          status: 200,
+          headers: {
+            'x-ms-request-id': '774c96e7-0001-0006-7e01-67617f000000',
+            'x-ms-version': '2016-05-31',
+            date: new Date().toISOString()
+          },
+          body: {
+            name: diskName,
+            location: loc,
+            zones: [zone],
+            sku: sku,
+            properties: {
+              provisioningState: 'Succeeded',
+              creationData: {
+                createOption: 'Copy',
+                sourceResourceId: 'customResource'
+              }
+            }
+          }
+        });
+        return client.createDiskFromSnapshot(snapshotName, zone, {
+            sku: sku,
+            tags: {
+              name: 'value'
+            }
+          })
+          .then((result) => {
+            expect(result.volumeId).to.equal(diskName);
+            expect(result.zone).to.equal(zone);
+            expect(result.type).to.equal('Premium_LRS');
```

Comment:
Please verify all keys in response e.g. size, extra etc.

Reply:
Done in latest commit